src/yaml/from_yaml.cr

```diff
@@ -21,16 +21,23 @@ def Array.from_yaml(string_or_io)
 end
 
 def Nil.new(pull : YAML::PullParser)
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_scalar
+  when "", "NULL", "Null", "null"
```

Comment:
It should, but it doesn't I will have to look more in depth to libyaml again tonight. But I was running into issues where when I inspect the return value of `read_scalar` I get `"null"`.